### ISSUE TITLE
Logic Analyzer: results on all 4 channels are shown even though only one channel is physically connected

### ISSUE THREAD

**Issue by marcnause** [fossasia/pslab-android#2523]
**Actual Behaviour**

The Logic Analyzer displays results for all 4 channels, even though only LA1 is physically connected to a source.

**Expected Behaviour**

The Logic Analyzer only shows a result for channels which are connected to a source.

**Steps to reproduce it**

1. Connect LA1 to Square Wave 4
2. Open app (if it has not opened automatically
3. Select Wave Generator
4. Set frequency and duty cycle for Square Wave 4
5. Press "play" symbol on top of screen
6. Select Logic Analyzer
7. Change Channel Selection to 4
8. Press "Analyze" button

**Screenshots of the issue**

![Screenshot_20240804-222355](https://github.com/user-attachments/assets/5a027525-de0b-4a5d-9a30-b82ad3d05398)

**Would you like to work on the issue?**

Maybe

**Additional information**

 I tested with PSLab v5 with firmware version 3.0.2.

**Comment by marcnause** [comment]
I tested again. I selected 4 channels.

If connected to `LA1` I get the same result as described above.
If connected to `LA2` I get a result for `LA2` only.
If connected to `LA3` I get a result for `LA3` only.
If connected to `LA4` I get no result.

I also tried with 3 channels.
If connected to `LA1` I get results on `LA1`, `LA2` and `LA3`
If connected to `LA2` I get a result for `LA2` only.
If connected to `LA3` I get no result.
